docs/source/technical-guide/command-line/publish-cs.rst

```diff
@@ -1,2 +1,472 @@
 publish-cs
 ~~~~~~~~~~
+Generates c# classes for all ItemTypes and ReusableTypes in the model. The generated classes can be serialized to json, populated from json, and serialized to XML.
+
+Command Line Arguments
+----------------------
+Required inputs for publish-cs command (must be specified in order).
+
+* ``[CogsLocation]`` 
+
+    The location of the folder containing model.
+
+* ``[TargetLocation]`` 
+
+    The location of the folder where the output will be created.
+
+Command Line Flags
+----------------------
+Optional inputs for publish-cs command.
+
+* ``-?|-h|--help``
+
+    Displays all command arguments and flags are for the publish-cs command.
+
+* ``-o|--overwrite``
+
+    If the ``[TargetLocation]`` is not empty, erase all files in the folder before generation.
+
+* ``-n|--namespace``
+
+    Allows user to specify the XMI of desired XML namespace used in XML creation from generated c# classes.
+
+* ``-p|--prefix``
+
+    Allows user to specify the prefix for XML namespace used in XML creation from generated c# classes.
+
+Example Command Line Usage
+--------------------------
+A few examples of how the command line arguments and flags can be used together.
+
+.. code-block:: console
+
+    publish-cs -h
+    publish-cs C:\Users\kevin\Documents\GitHub\cogs\cogsburger C:\Users\kevin\Documents\GitHub\cogs\Cogs.Console\out
+    publish-cs -o C:\Users\kevin\Documents\GitHub\cogs\cogsburger C:\Users\kevin\Documents\GitHub\cogs\Cogs.Console\out
```

Comment:
lets use relative paths and generic descriptive names in all of the examples, such as

publish-cs -o "MyCogsModelDirectory" "MyOutputDirectory"